tool/lib/constants-sampler.js

```diff
@@ -114,6 +114,12 @@ module.exports = class ConstantSampler {
                                 constants.push([`param:@${device}.${f}:${arg}:Entity(${entityType})`, `null`, sample]);
                             });
                         }
+
+                        if (arg === 'id') {
+                            samples.forEach((sample) => {
+                                constants.push([`param:@${device}.${f}:${arg}:String`, sample]);
+                            });
+                        }
```

Comment:
What is the bug exactly here? (That is, can you expand the commit message?)

Reply:
We use string value instead of entity value when filtering on id...

Reply:
Oh I see.